pkg/base/node_id.go

```diff
@@ -95,21 +104,21 @@ const TempStoreID = -1
 // stores if they haven't been initialized. If a main store hasn't
 // been initialized, then "?" is returned.
 func (s *StoreIDContainer) String() string {
-	return redact.StringWithoutMarkers(s)
-}
-
-// SafeFormat implements the redact.SafeFormatter interface.
-func (s *StoreIDContainer) SafeFormat(w redact.SafePrinter, _ rune) {
 	val := s.Get()
 	if val == 0 {
```

Comment:
I'd move all of this logic into `Set` or you can have races where `val != 0` but `str` is still nil.